charms/jupyter-ui/tests/unit/test_operator.py

```diff
@@ -594,3 +592,26 @@ def test_failure_get_config(
 
         # Assert
         harness.charm.logger.warning.assert_called_once()
+
+    @patch("charm.KubernetesServicePatch", lambda x, y, service_name: None)
+    @patch("charm.JupyterUI.k8s_resource_handler")
+    def test_sidecar_and_ambient_relations_added(
+        self, k8s_resource_handler: MagicMock, harness: Harness
+    ):
+        """Test the charm is in BlockedStatus when both sidecar and ambient relations are added."""
+        # Arrange
+        harness.add_relation("ingress", "istio-pilot")
+
+        harness.add_relation("istio-ingress-route", "istio-ingress-k8s")
```

Comment:
suggestion (non-blocking): do we want to import some names from Chisme, to avoid hardcoding them into each charm's repository?